test/signs_ui/config/expiration_test.exs

```diff
@@ -50,12 +50,139 @@ defmodule SignsUi.Config.ExpirationTest do
         }
       }
 
-      assert SignsUi.Config.Expiration.expire_signs(state, fn ->
+      assert SignsUi.Config.Expiration.expire_signs_via_time(state, fn ->
                DateTime.new!(~D[2019-01-15], ~T[08:00:00], "America/New_York")
              end) == expected_updates
     end
   end
 
+  describe "expire_signs_via_alert/2" do
+    test "produces correct updates" do
+      alert_state = %SignsUi.Alerts.State{
+        alerts: %{
+          "Red" => %{
+            "1234" => %{},
+            "5678" => %{}
+          },
+          "Blue" => %{
+            "abc" => %{}
+          }
+        }
+      }
+
+      alert_ids = SignsUi.Config.Expiration.active_alert_ids(alert_state)
+      assert alert_ids == MapSet.new(["1234", "5678", "abc"])
+
+      sign_state = [
+        %SignsUi.Config.Sign{
+          id: "250",
+          config: %{mode: :auto}
+        },
+        %SignsUi.Config.Sign{
+          id: "460",
+          config: %{
+            mode: :headway,
+            expires: nil,
+            alert_id: nil
+          }
+        },
+        %SignsUi.Config.Sign{
+          id: "898",
+          config: %{
+            mode: :headway,
+            expires: nil,
+            alert_id: "1234"
+          }
+        },
+        %SignsUi.Config.Sign{
+          id: "925",
+          config: %{
+            mode: :headway,
+            expires: nil,
+            alert_id: "326"
+          }
+        },
+        %SignsUi.Config.Sign{
+          id: "793",
+          config: %{
+            mode: :off,
+            expires: nil,
+            alert_id: nil
+          }
+        },
+        %SignsUi.Config.Sign{
+          id: "367",
+          config: %{
+            mode: :off,
+            expires: nil,
+            alert_id: "5678"
+          }
+        },
+        %SignsUi.Config.Sign{
+          id: "471",
+          config: %{
+            mode: :off,
+            expires: nil,
+            alert_id: "437"
+          }
+        },
+        %SignsUi.Config.Sign{
+          id: "714",
+          config: %{
+            mode: :static_text,
+            line1: "test",
+            line2: "test",
+            expires: nil,
+            alert_id: nil
+          }
+        },
+        %SignsUi.Config.Sign{
+          id: "474",
+          config: %{
+            mode: :static_text,
+            line1: "test",
+            line2: "test",
+            expires: nil,
+            alert_id: "abc"
+          }
+        },
+        %SignsUi.Config.Sign{
+          id: "273",
+          config: %{
+            mode: :static_text,
+            line1: "test",
+            line2: "test",
+            expires: nil,
+            alert_id: "593"
+          }
+        }
+      ]
+
+      expired_signs = SignsUi.Config.Expiration.expire_signs_via_alert(sign_state, alert_ids)
+
+      expected_signs = [
+        %SignsUi.Config.Sign{
+          id: nil,
```

Comment:
These are supposed to be the sign IDs, not the alert IDs, of the signs that are getting expired (i.e.: signs that have an alert ID that is not present in the list of active alerts).